### ISSUE TITLE
minTimeUUID/maxTimeUUID not implemented

### ISSUE THREAD

**Issue by Gapmeister66** [YugaByte/yugabyte-db#573]
I found the following error using Java Driver 3.2.0 against YBDB 1.1.4.0
      
1) Error injecting constructor, com.datastax.driver.core.exceptions.InvalidQueryException: SQL error: Invalid Function Call. Failed calling 'mintimeuuid(anytype)'. Not implemented: Builtin function 'mintimeuuid' is not yet implemented
SELECT * FROM content_watch_snapshot WHERE day IN ? AND snapshottime >= minTimeuuid(?);
                                                                        ^^^^^^^^^^^

We're reporting this error as we are evaluating YBDB and would like to see this issue fixed.

**Comment by kmuthukk** [comment]
@Gapmeister66 : thanks for reporting. Will get back to you quickly with an estimate.

hi @m-iancu , @nocaway 

Looks like we have not implemented minTimeUuid() and maxTimeUuid(). 

```
template<typename PTypePtr, typename RTypePtr>
CHECKED_STATUS ConvertToMinTimeuuid(PTypePtr source, RTypePtr target) {
  return STATUS(RuntimeError, "Not yet implemented");
}
```
Thoughts on what's involved? And any workaround suggestions for them?

**Comment by m-iancu** [comment]
@Gapmeister66 We should have a fix ready soon (and hopefully a release with it in a week or so). 

In the meantime, if this is urgent, a possible workaround can be to implement the function in the app. i.e. change the query to:
```
SELECT * FROM content_watch_snapshot WHERE day IN ? AND snapshottime >= ?;
```
And bind the max timeuuid as computed in the app for the second value.

**Comment by m-iancu** [comment]
@Gapmeister66 Fixed in 234e23c and should be available in the next (minor) release. 

Note that, like in Cassandra, the TimeUUIDs generated by `minTimeUUID/maxTimeUUID` are not "real" TimeUUIDs as they do not follow the generation process specified in RFC 4122. They are intended to be used for comparisons (like in your example use case) and not to be e.g. inserted in the database. 
For these intended use-cases the two functions should work exactly like in Cassandra.

However, for completeness, also note that the TimeUUID ordering in YugaByte differs slightly from Cassandra. For the timestamp component (most-significant-bytes) it should indeed compare exactly like Cassandra. But, for the node/clock component (least-significant-bytes) Cassandra compares the values as signed bytes while we compare as unsigned based on [RFC-4122#4.1.2](https://tools.ietf.org/html/rfc4122#section-4.1.2) definition. Our `minTimeUUID/maxTimeUUID` accounts for that so, in practice, the comparison behavior should be exactly the expected one -- but the actual values return will be different from the ones in Cassandra (for the least-significant-bytes only).
